Sources/JWTKeychain/Support/Configuration/Configuration.swift

```diff
@@ -15,67 +15,103 @@ public protocol ConfigurationType {
-    
+
     /// Key used to sign the JWT
     public var signatureKey: String
-    
+
     /// Key used to check the signing the JWT
     public var publicKey: String? = nil
-    
+
     /// Which signer will be used while signing the JWT
     public var signer: String
 
-    private init(signer: String, signatureKey: String, publicKey: String?, secondsToExpire: Double){
+    public enum Error: Swift.Error {
+        case noJWTConfig
+        case missingConfig(String)
+    }
+
+    public init(drop: Droplet) throws {
+        try self.init(config: drop.config)
+    }
+
+    public init(config: Config) throws {
+
+        guard let jwtConfig = config["jwt"]?.object else {
+            throw Error.noJWTConfig
+        }
+
+        guard let signer: String = jwtConfig["signer"]?.string else {
+            throw Error.missingConfig("signer")
+        }
+
+        guard let secondsToExpire = jwtConfig["secondsToExpire"]?.double else {
+            throw Error.missingConfig("secondsToExpire")
+        }
+
+        guard let signatureKey = jwtConfig["signatureKey"]?.string else {
+            throw Error.missingConfig("signatureKey")
+        }
+
+        let publicKey: String? = jwtConfig["publicKey"]?.string
+
+        if publicKey == nil {
+            //The ECDSA and RSA (ES*/RS*) signers take a private key for signing and needs a matching public key for verifying.
+            if signer.hasPrefix("ES") || signer.hasPrefix("RS"){
+                throw Error.missingConfig("publicKey")
+            }
+        }
+
+        self.init(signer: signer, signatureKey: signatureKey, publicKey: publicKey, secondsToExpire: secondsToExpire)
+
+    }
+
+    public init(signer: String, signatureKey: String, publicKey: String?, secondsToExpire: Double){
         self.signer = signer
         self.signatureKey = signatureKey
         self.publicKey = publicKey
-        self.secondsToExpire = secondsToExpire}
-    // Register configs
-    public static func boot(signer: String, signatureKey: String, publicKey: String?, secondsToExpire: Double) throws {
-        Configuration.instance = Configuration(signer: signer, signatureKey: signatureKey, publicKey: publicKey, secondsToExpire: secondsToExpire)
-    }
-    
-    
+        self.secondsToExpire = secondsToExpire
+      }
+
     /// Gets token signature key
     ///
     /// - Returns: signature key
-    public static func getTokenSignatureKey() -> Bytes {
-        
-        return Array(Configuration.instance!.signatureKey.utf8)
-        
+    public func getTokenSignatureKey() -> Bytes {
+        return Array(self.signatureKey.utf8)
+
     }
-    
+
     /// Gets token public key
     ///
     /// - Returns: public key
     /// - Throws: if cannot retrieve signature key
-    public static func getTokenPublicKey() throws -> Bytes {
-        return Array(Configuration.instance!.publicKey!.utf8)
-        
+    public func getTokenPublicKey() throws -> Bytes {
+        return Array(self.publicKey!.utf8)
```

Comment:
Even though there is a nil check, the method is public, so if someone calls it from outside this module, it will crash. I think what should be done here is that it should be completely changed  :D 

As in, there's no need for it to be a function - it should be a var - and also the naming is not very Swift-y. 

So I'd suggest something like this:

```swift
var publicKeyBytes: Bytes? {
    if let publicKey = publicKey {
       return Array(publicKey.utf8)
    } else {
       return nil 
    }
}
```

Then in the place where there was the nil check originally you'd just unwrap it in an if let, like so:

```swift
if let publicKeyBytes = publicKeyBytes {
    key = publicKeyBytes
}
```

Now that I come to think of it, same change should apply to the signature key (function -> variable).